services/grpcService.ts

```diff
@@ -246,19 +269,54 @@ export class GrpcBlockchainClient implements BlockchainClient {
 
       // output for data stream
       void confirmedStream.on('data', (data: TransactionNotification) => {
-        const objectTxn = data.getConfirmedTransaction()!.toObject()
-        console.log(`${nowDateString}: got confirmed txn`, objectTxn)
-        onTransactionNotification(objectTxn)
+        void this.processSubscribedNotification(data)
       })
-      void unconfirmedStream.on('data', (data: TransactionNotification) => {
-        const unconfirmedTxn = data.getUnconfirmedTransaction()!.toObject()
-        const objectTxn = this.parseMempoolTx(unconfirmedTxn)
-        console.log(`${nowDateString}: got unconfirmed txn`, objectTxn)
-        onMempoolTransactionNotification(objectTxn)
+      // subscribed addresses
+      value.forEach(address => {
+        this.subscribedAddresses[address.address] = address
       })
+    }
+  }
+
+  private async getPrismaTransactionsForSubscribedAddresses (transaction: Transaction.AsObject, confirmed: boolean): Promise<AddressWithTransaction[]> {
+    const addressWithTransactions: AddressWithTransaction[] = await Promise.all(Object.values(this.subscribedAddresses).map(
+      async address => {
+        return {
+          address,
+          transaction: await this.getTransactionFromGrpcTransaction(transaction, address, confirmed)
+        }
+      }
+    ))
+    return addressWithTransactions.filter(
+      addressWithTransaction => addressWithTransaction.transaction.amount > new Prisma.Decimal(0)
+    )
+  }
+
+  private async processSubscribedNotification (data: TransactionNotification): Promise<void> {
+    let addressWithConfirmedTransactions: AddressWithTransaction[] = []
+    let addressWithUnconfirmedTransactions: AddressWithTransaction[] = []
 
-      console.log(`${nowDateString}: txn data stream established for addresses ${addresses.join(', ')}.`)
+    // get new confirmed transactions
+    const confirmedTransaction = data.getConfirmedTransaction()?.toObject()
+    if (confirmedTransaction != null) {
+      addressWithConfirmedTransactions = await this.getPrismaTransactionsForSubscribedAddresses(confirmedTransaction, true)
+
+      // remove unconfirmed transactions that have now been confirmed
+      const transactionsToDelete = await fetchUnconfirmedTransactions(confirmedTransaction.hash as string)
+      await deleteTransactions(transactionsToDelete)
     }
-    await createTxnStream()
-  };
+
+    // get new unconfirmed transactions
+    const unconfirmedTransaction = data.getUnconfirmedTransaction()?.toObject()
```

Comment:
AFAIK this won't ever get called as `processSubscribedNotification` is only being called on the `confirmedTransaction` stream. Maybe refactor this to accept a `confirmed` parameter?

Reply:
fix in the comment above solves this